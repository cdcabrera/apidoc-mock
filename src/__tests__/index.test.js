const { apiDocMock, setupResponse } = require('../');
const { OPTIONS } = require('../global');

describe('ApiDocMock', () => {
  it('should have specific defined properties', () => {
    expect(apiDocMock).toBeDefined();
    expect(setupResponse).toBeDefined();
  });

  it('should have support functions that fail gracefully', () => {
    expect(setupResponse()).toBe(null);
  });

  it('should throw an error during testing', async () => {
    const func = async () => apiDocMock();
    await expect(func).rejects.toThrow('Server failed to load');
  });

  it('should generate an api response from annotations', async () => {
    const apiFixture = generateFixture(
      `/**
         * @api {get} /hello/world/
         * @apiSuccessExample {json} Success-Response:
         *     HTTP/1.1 200 OK
         *     {
         *       "success": "test"
         *     }
         * @apiErrorExample {json} Error-Response:
         *     HTTP/1.1 400 OK
         *     {
         *       "error": "test"
         *     }
         */`,
      { dir: './.fixtures/apiDocMock', filename: 'apiDocMock.js' }
    );

    const { mockClear } = mockObjectProperty(OPTIONS, {
      watchPath: [apiFixture.dir],
      docsPath: 'lorem-ipsum'
    });

    const { apiJson, ...rest } = await apiDocMock();
    expect({
      apiJson: JSON.stringify(
        apiJson.map(({ error, success }) => ({ errorExamples: error?.examples, successExamples: success?.examples })),
        null,
        2
      ),
      ...rest
    }).toMatchSnapshot('apiDocMock, parsed');

    mockClear();
  });
});
