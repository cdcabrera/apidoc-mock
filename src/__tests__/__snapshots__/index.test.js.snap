// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiDocMock should generate an api response from annotations: apiDocMock, parsed 1`] = `
{
  "CACHE": {
    "app": {
      "delete": [MockFunction],
      "get": [MockFunction] {
        "calls": [
          [
            "/hello/world/",
            [Function],
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "listen": [MockFunction] {
        "calls": [
          [
            undefined,
            [Function],
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "patch": [MockFunction],
      "post": [MockFunction],
      "put": [MockFunction],
      "use": [MockFunction] {
        "calls": [
          [
            "/docs",
            undefined,
          ],
          [
            [Function],
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
          {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
    "httpTerminator": {
      "terminate": [Function],
    },
  },
  "apiJson": "[
  {
    "errorExamples": [
      {
        "title": "Error-Response:",
        "content": "HTTP/1.1 400 OK\\n{\\n  \\"error\\": \\"test\\"\\n}",
        "type": "json"
      }
    ],
    "successExamples": [
      {
        "title": "Success-Response:",
        "content": "HTTP/1.1 200 OK\\n{\\n  \\"success\\": \\"test\\"\\n}",
        "type": "json"
      }
    ]
  }
]",
  "httpTerminator": {
    "terminate": [Function],
  },
}
`;
